test/datomish/test/query.cljc

```diff
@@ -684,3 +684,95 @@
                                    "something")
                              '[[?save]]]]}
                   conn)))))
+
+(deftest-db test-find-specs-expansion conn
+  (let [attrs (<? (<initialize-with-schema conn save-schema))]
+    ;; Relation.
+    (is (= {:select (list [:fulltext_datoms0.v :title])
+            :modifiers [:distinct]
+            :from (list [:fulltext_datoms 'fulltext_datoms0])
+            :where (list :and [:= :fulltext_datoms0.a (:save/title attrs)])}
+           (expand [:find '?title
+                    :in '$
+                    :where '[?save :save/title ?title]]
+                   conn)))
+
+    ;; Tuple. We expect only one result, and indeed we only take one.
+    ;; No need for :distinct in this case!
+    (is (= {:select (list [:fulltext_datoms0.v :title])
+            :modifiers []
+            :limit 1
+            :from (list [:fulltext_datoms 'fulltext_datoms0])
+            :where (list :and [:= :fulltext_datoms0.a (:save/title attrs)])}
+           (expand [:find '[?title]
+                    :in '$
+                    :where '[?save :save/title ?title]]
+                   conn)))
+
+    ;; Scalar. As with the tuple form, we expect only one result.
+    (is (= {:select (list [:fulltext_datoms0.v :title])
+            :modifiers []
+            :limit 1
+            :from (list [:fulltext_datoms 'fulltext_datoms0])
+            :where (list :and [:= :fulltext_datoms0.a (:save/title attrs)])}
+           (expand [:find '?title '.
+                    :in '$
+                    :where '[?save :save/title ?title]]
+                   conn)))
+
+    ;; Collection.
+    (is (= {:select (list [:fulltext_datoms0.v :title])
+            :modifiers [:distinct]
+            :from (list [:fulltext_datoms 'fulltext_datoms0])
+            :where (list :and [:= :fulltext_datoms0.a (:save/title attrs)])}
+           (expand [:find '[?title ...]
+                    :in '$
+                    :where '[?save :save/title ?title]]
+                   conn)))))
+
+(defn orderless=
+  "Compare two arrays regardless of order."
+  [a b]
+  (= (set a) (set b)))
+
+(deftest-db test-find-specs-result-shape conn
+  (let [attrs (<? (<initialize-with-schema conn save-schema))]
+    ;; Add some data.
+    (<? (d/<transact! conn
+                      [{:db/id (d/id-literal :db.part/user -1)
+                        :save/title "Some page title"}
+                       {:db/id (d/id-literal :db.part/user -2)
+                        :save/title "A different page"}]))
+
+    ;; Relation.
```

Comment:
If you mean changing our return value into a set: Datomic doesn't do it, and it's typically needless expense. If you mean something else, could you clarify?